### ISSUE TITLE
Allow preset quotation styles, localised to the current Project's language

### ISSUE THREAD

**Issue by tmarplatt** [vkbo/novelWriter#1772]
**The problem**

novelWriter, by default, is configured to match the English quotation style. Using novelWriter for writing in other languages means needing to reconfigure the quotation style.

Currently, this means all users that write in a different language have to enter this configuration on their own. We are expecting these users to be familiar with the quotation style features, to understand their potential and depth, and to perform the necessary configuration (e.g. see #1422).

Across so many novelWriter installations, users around the world are (once they've become aware of it) entering the same configuration themselves: identical each and every time for the same language. Needlessly. It's a cumulated and avoidable waste of time and effort.

Ideally, all a user should do is configure the Project language, and novelWriter would apply the formal quotation configuration for the target language.

**The solution**

- Allow contributors to submit quoting presets and maintain one per language.
- Each preset is a configuration scheme for the different quoting options: 
  - opening and closing quotes
  - quote highlighting
  - quote conversion
- Ship novelWriter with these presets built-in.
- When a user configures a new Project, check if the chosen Project language matches an existing preset, and change the quotation preferences accordingly, automatically, in a transparent way. The user needs not intervene.

**Comment by tmarplatt** [comment]
If #1770 was implemented, the preset would also include a custom number of quotation nesting levels. Similar case for #1771.

**Comment by vkbo** [comment]
> Across so many novelWriter installations, users around the world are (once they've become aware of it) entering the same configuration themselves: identical each and every time for the same language. Needlessly. It's a cumulated and avoidable waste of time and effort.

Cumulated effort doesn't matter. It's a small config job for each user. Most software has such a set of rules to configure.

> Ideally, all a user should do is configure the Project language, and novelWriter would apply the formal quotation configuration for the target language.

That makes the erroneous assumption that there is only one configuration for each language, which there isn't. So we need a separate configuration of this anyway. That's precisely why the user can pick the symbols directly. It allows for more flexibility.

> * Allow contributors to submit quoting presets and maintain one per language.

That could in principle be done. However, it would be very annoying to users without their particular language setting available until it has been added. Thus we cannot remove the current custom implementation, and end up having two ways to configure quotes. This is confusing and complicated.

> * When a user configures a new Project, check if the chosen Project language matches an existing preset, and change the quotation preferences accordingly, automatically, in a transparent way. The user needs not intervene.

As mentioned earlier, it's not that simple.

**Comment by vkbo** [comment]
Given this some thought too. I think instead of a et of fixed rules per language, having a set of flexible settings with a dropdown of predefined rules that when selected will set the correct default settings for that language is the way to go. This would require contributions from users, but some way to export as a JSON file and submit it may solve that.

**Comment by vkbo** [comment]
I think a good solution here would be to add a new settings tool for auto-formatting and dialogue and take them out of main preferences, but leave a button there to access them.

The settings would instead be a set of editor preferences that have a default and the option to override on a per-project basis. I'm thinking of adding a settings button directly in the editor where these can be accessed.

I'm not sure where to keep the settings. If I put them in their own form, I can in principle reuse the form in Preferences for the default setting, and also add them in Project Settings for override. Then the button in the editor would just pop the correct page in Project Settings. Alternatively, I can create a brand new setting tool and put it in the menu as "Editor Settings".

I'll have to experiment a little with what works. I would prefer to only have these settings in a single place anyway, and just add a switch to enable project level override.

**Comment by tmarplatt** [comment]
Yes the whole specificity of auto formatting, quoting rules, etc., justifies having separate controls.

> The settings would instead be a set of editor preferences that have a default and the option to override on a per-project basis.

How do you integrate that with this?

> predefined rules that when selected will set the correct default settings for that language is the way to go.

**Comment by vkbo** [comment]
> How do you integrate that with this?

My thinking was that you can select a set and it will fill in the relevant values, as a sort of template. You are still free to change them. This is important, because there are a ton of variations here, and likely also personal styles and preferences. These settings are highlighting for visualisation, and has no effect on formatting beyond that, so there needs to be full flexibility.

If I move them all to a separate settings dialog, I would have a switch to say if they were global or just for the current project.

**Comment by tmarplatt** [comment]
Thinking from the perspective of first time users, it's better not to show any highlighting than to show incorrect/unexpected highlighting for the language being used. Until the user realizes these controls exist, they may just presume the language is not supported by the highlighting system. This, in my opinion, is damaging because of the disappointment it causes.

On the flip side, it's a rare treat to find out software properly supports your non English language in all of its nuances. So it becomes an issue of visibility.

**Comment by vkbo** [comment]
> Thinking from the perspective of first time users, it's better not to show any highlighting than to show incorrect/unexpected highlighting for the language being used.

Agreed. I was not planning on tying this to the selected project language. More having a combo box labelled "Load presets for" followed by "Portuguese, Spanish, Polish, ..." and so on, and let users contribute the settings for more languages.

> Until the user realizes these controls exist, they may just presume the language is not supported by the highlighting system. This, in my opinion, is damaging because of the disappointment it causes.

The settings currently default to highlight double quotes, and nothing else. I think that's a sensible starting point, but I'm thinking of collecting all the language-based settings in one place that is easily reachable. Maybe from a button directly in the editor. It can also auto-pop when you create a new project and open it for the first time.

> On the flip side, it's a rare treat to find out software properly supports your non English language in all of its nuances. So it becomes an issue of visibility.

Well, English is not my first language, so it was added early on, even if I didn't originally add a translation framework. I tend to use software in English, because Norwegian GUI terminology is so corny.

**Comment by tmarplatt** [comment]
When it comes to GUI labelling itself I believe most languages lack in appropriate lexicon and most of it is borrowed from English anyway. But in this case it's a feature that places the uses of your native language front and center, which is where the user feels the impact of good language support.

The Epic refactor is such an exciting feature for this alone. And I am looking forward to it.

**Comment by vkbo** [comment]
Icelandic is a language that has a very strong focus on preserving it and adding new words when needed instead of just mimicking English. I can understand it a little (seeing that it is derived from Norse and all) and they have so many great words.
